enhancements/ingress/ingress-fault-detection.md

```diff
@@ -0,0 +1,174 @@
+external traffic bound for application pods is unable to reach (or experiences difficulties en-route to) the correct application destination.
+
+## Motivation
+
+Cluster Ingress issues can greatly and rapidly impact the health of an OpenShift cluster. Cluster Administrators should be alerted to
+any potential ingress complications via the Ingress Operator as soon as they occur, as to help minimize unplanned application downtime.
+OpenShift cluster administrators should not first find out about ingress breakages when they inspect the status of a different critical
+cluster component, such as the console. A Cluster Administrator should also have tooling available to them to thoroughly diagnose and
+understand why routes on their cluster are not working.
+
+This enhancement also has benefits for internal OpenShift development.
+Often times when networking-related bugs are identified within a cluster, determining which networking sub-component
+(router, SDN, DNS, etc.) is responsible for the observed bug is difficult. Bug reports tend to spend lots of time bouncing between different teams
+as engineers are not sure which component(s) is legitimately at fault. Being able to quickly pinpoint Ingress related issues will expedite the bugfix process.
+
+### Goals
+
+1. Teach the [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) to automatically detect cluster Ingress issues
+and alert cluster administrators when relevant problems are discovered.
+2. Provide a secondary tool that can be triggered at will by a Cluster Administrator or an OpenShift support engineer that spawns an Ingress "debug"
+container that quickly identifies and triages potential Ingress issues within the cluster.
+
+### Non-Goals
+
+1. Teach the Ingress Operator to "self-heal" when cluster Ingress errors or issues are observed.
+2. Modify the [Ingress Controller API](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)
+to allow Cluster Administrators to disable the periodic Ingress Operator fault detector at will for a given Ingress Controller.
+
+## Proposal
+
+Once the Ingress Operator has successfully deployed an [Ingress Controller](https://github.com/openshift/cluster-ingress-operator/blob/master/README.md)
+within a cluster, the Ingress Operator will spawn a simple HTTP echo container, such as [Hello Openshift](https://hub.docker.com/r/openshift/hello-openshift/).
+The Ingress Operator will also create the necessary Service and Route resources to provide Ingress to the echo container within the `openshift-ingress-operator`
+namespace. Then, the Ingress Operator will periodically attempt to send a request to the echo container, and await a proper response. These requests will be sent
+so that they are routed outside of the cluster to the cloud load balancer first, in order to ensure that each peice of the Ingress "chain" is being tested.
```

Comment:
We should also ensure that we cover HTTP/2 enabled routes (where we can) and will therefore need a custom certificate.

Reply:
mentioned. Thanks!